pkg/meta/meta.go

```diff
@@ -1038,16 +1038,27 @@ func (m *Meta) GetMetasByDBID(dbID int64) ([]structure.HashPair, error) {
 	return res, nil
 }
 
+var checkSubstringsInOrder = [4]string{
+	`"partition":null`,
+	`"tiflash_replica":null`,
+	`"policy_ref_info":null`,
+	`"ttl_info":null`,
+}
+
 // CheckSpecialAttributes checks if the special attributes are in the table info.
 // Make it same as hasSpecialAttributes.
 // Exported for testing.
 // It's the regexp version for hasSpecialAttributes(), please keep up-to-date with it.
-func CheckSpecialAttributes(str string) bool {
-	if strings.Contains(str, "\"tiflash_replica\":null") && strings.Contains(str, "\"policy_ref_info\":null") &&
-		strings.Contains(str, "\"partition\":null") && strings.Contains(str, "\"ttl_info\":null") {
-		return false
+func CheckSpecialAttributes(json []byte) bool {
+	idx := 0
+	for _, substr := range checkSubstringsInOrder {
+		idx = bytes.Index(json, hack.Slice(substr))
+		if idx == -1 {
+			return true
+		}
+		json = json[idx:]
```

Comment:
are they always appear in `checkSubstringsInOrder` order?

Reply:
TestCheckSpecialAttributes have checked the order, this is the order of fields appear in TableInfo. I see the doc does not describe the behaviour, only the internal function shows the marshalling key is ordered by field index.

If this logic is wrong, we will fallback to the slow path.